src/Airship/Internal/Decision.hs

```diff
@@ -242,8 +242,11 @@ b04 r@Resource{..} = do
 b03 r@Resource{..} = do
     trace "b03"
     req <- lift request
+    allowed <- lift allowedMethods
     if requestMethod req == HTTP.methodOptions
-        then lift $ halt HTTP.status200
+        then do
+            lift $ addResponseHeader ("Allow",  intercalate "," allowed)
+            lift $ halt HTTP.status200
```

Comment:
I don't really know tbh. The spec is pretty vague. All it says is:

```
A 200 response SHOULD include any header fields that indicate
   optional features implemented by the server and applicable to that
   resource (e.g., Allow)...
```
I found all of two sites in the wild from a quick and unscientific test that accept OPTIONS requests, but they both return 200. That of course doesn't mean it's correct and given the language in the spec it sounds like we have some leeway so I'm open to 200 or 204:

```
 curl -v -X OPTIONS http://www.reddit.com
* Rebuilt URL to: http://www.reddit.com/
*   Trying 198.41.209.139...
* Connected to www.reddit.com (198.41.209.139) port 80 (#0)
> OPTIONS / HTTP/1.1
> Host: www.reddit.com
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Date: Fri, 31 Jul 2015 16:47:45 GMT
< Content-Type: text/html; charset=UTF-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< Set-Cookie: __cfduid=d67c1feb7091b0f82a1b087f05630ee5e1438361265; expires=Sat, 30-Jul-16 16:47:45 GMT; path=/; domain=.reddit.com; HttpOnly
< x-ua-compatible: IE=edge
< x-frame-options: SAMEORIGIN
< x-content-type-options: nosniff
< x-xss-protection: 1; mode=block
< cache-control: max-age=0, must-revalidate
< X-Moose: majestic
< Server: cloudflare-nginx
< CF-RAY: 20eac9731a84202a-DFW
< 
* Connection #0 to host www.reddit.com left intact
```

```
 curl -v -X OPTIONS http://www.cnn.com
* Rebuilt URL to: http://www.cnn.com/
*   Trying 23.235.44.73...
* Connected to www.cnn.com (23.235.44.73) port 80 (#0)
> OPTIONS / HTTP/1.1
> Host: www.cnn.com
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< x-servedByHost: prd-10-60-170-45.nodes.56m.dmtio.net
< Cache-Control: max-age=60
< Allow: GET,HEAD
< Content-Type: text/html; charset=utf-8
< Accept-Ranges: bytes
< Via: 1.1 varnish
< Content-Length: 8
< Accept-Ranges: bytes
< Date: Fri, 31 Jul 2015 16:50:49 GMT
< Via: 1.1 varnish
< Connection: keep-alive
< X-Served-By: cache-iad2151-IAD, cache-dfw1827-DFW
< X-Cache: MISS, MISS
< X-Cache-Hits: 0, 0
< X-Timer: S1438361449.638794,VS0,VE65
< Vary: Accept-Encoding
< 
* Connection #0 to host www.cnn.com left intact
GET,HEAD
```